src/AddTodoForm.jsx

```diff
@@ -1,32 +1,26 @@
-
-
 import React, { useState } from 'react';
 
-const AddTodoForm = ({ onAddTodo }) => {
-
-  const [todoTitle, setTodoTitle] = React.useState('');
-
-  // new Function
-  const handleAddTodo = (event) => {
-    // hint
-    event.preventDefault();
+function AddTodoForm({ onAddTodo }) {
+  const [todoTitle, setTodoTitle] = useState('');
 
-    const todoTitle = event.target.title.value;
-
-    console.log(todoTitle);
-
-    event.target.reset();
-
-    // last sentence of the part that says app,invoke the onAddTodo
-    onAddTodo(todoTitle);
+  const handleTitleChange = (event) => {
+    setTodoTitle(event.target.value);
+  };
 
+  const handleAddTodo = () => {
+    const newTodo = {
+      id: Date.now(), //identificador único temporal
+      title: todoTitle
+    };
+    onAddTodo(newTodo);
+    setTodoTitle('');
   };
 
   return (
-    <form onSubmit={handleAddTodo}>
-      <input type="text" name="title" />
-      <button type="submit">Add</button>
-    </form>
+    <div>
```

Comment:
I have a question, I know that I am learning React, to validate the forms do I have to do it with js or can I use html5 attributes? as "required" or do I have to do both validations? example `requiered` `      <form onSubmit={handleAddTodo}> 
        <label htmlFor="todoTitle">Fill: </label>
        <input type="text" id="todoTitle" name="todoTitle" value={todoTitle} onChange={handleTitleChange} required/>
        <button type="submit">Add Todo</button>
      </form>  `

Reply:
You will eventually end up having many layers of form validation, from the frontend to server side. On the frontend using built-in in attributes such as `required` is a great start to build on. How robust your form validation is entirely up to you and often you will mix both html attributes and JS.

One example is, let's say you require an 8 character password. You can quickly implement the required constraint using HTML attributes and additionally you will need to check the length of the string provided. This will most likely be done working with React state and by writing out some code to check for the length of the string.